src/connection_wrap.cc

```diff
@@ -71,6 +73,46 @@ void ConnectionWrap<WrapType, UVType>::OnConnection(uv_stream_t* handle,
   wrap_data->MakeCallback(env->onconnection_string(), arraysize(argv), argv);
 }
 
+
+template <typename WrapType, typename UVType>
+void ConnectionWrap<WrapType, UVType>::AfterConnect(uv_connect_t* req,
+                                                    int status) {
+  ConnectWrap* req_wrap = static_cast<ConnectWrap*>(req->data);
+  CHECK_NE(req_wrap, nullptr);
+  WrapType* wrap = static_cast<WrapType*>(req->handle->data);
+  CHECK_EQ(req_wrap->env(), wrap->env());
+  Environment* env = wrap->env();
+
+  HandleScope handle_scope(env->isolate());
+  Context::Scope context_scope(env->context());
+
+  // The wrap and request objects should still be there.
+  CHECK_EQ(req_wrap->persistent().IsEmpty(), false);
+  CHECK_EQ(wrap->persistent().IsEmpty(), false);
+
+  bool readable, writable;
+
+  if (status) {
+    readable = writable = 0;
+  } else {
+    readable = uv_is_readable(req->handle) != 0;
+    writable = uv_is_writable(req->handle) != 0;
+  }
```

Comment:
I'm uncertain if this is alright for TCPWrap or not (TCPWrap always set these arguments to true) and wanted to open the PR to get some feedback.